### PR TITLE
Rewrite examples in documentation to TypeScript (#5873)

### PR DESCRIPTION
**Pull request by xnameTM**
## Summary
This pr adds TypeScript templates of code in the Reanimated Docs.

## How is it work?
I rewrote the `.jsx` exercises to `.tsx` and I added tsx to jsx code compilation in the InteractiveExample component.

## How to test this feature?
Command to change current working dictionary:
> cd docs

Command to download libraries:
> yarn install

Command to preview the project:
> yarn start

# Screenshots
Zrzut ekranu 2024-04-8 o 11 33
57 Zrzut ekranu 2024-04-8 o 11 34
26 Zrzut ekranu 2024-04-8 o 11 34
46

### CODE REVIEW

File: docs/src/examples/AnimatedReactionMeasure.tsx [hunk 1]
```diff
@@ -24,10 +24,7 @@ export default function App() {
     () => {
       const measurement = measure(animatedRef);
 
-      if (measurement !== null) {
-        const measuredWidth = parseInt(measurement.width, 10);
-        runOnJS(setText)(measuredWidth);
-      }
+      if (measurement !== null) runOnJS(setText)(measurement.width);
```
Comment (piaskowyk) [inline docs/src/examples/AnimatedReactionMeasure.tsx:27]:
```suggestion
      if (measurement !== null) {
        runOnJS(setText)(measurement.width);
       }
```

File: docs/src/examples/MeasureBasic.tsx [hunk 2]
```diff
@@ -9,17 +10,16 @@ import Animated, {
...
   const handlePress = () => {
     width.value = withTiming(width.value + 50, {}, () => {
       // highlight-next-line
-      const measurement = measure(animatedRef);
-      if (measurement === null) {
-        return;
-      }
+      const measurement: MeasuredDimensions | null = measure(animatedRef);
+
+      if (measurement === null) return;
```
Comment (piaskowyk) [inline docs/src/examples/MeasureBasic.tsx:22]:
```suggestion
      if (measurement === null) {
        return;
      }
```

File: docs/src/examples/SpringCarousel.tsx [hunk 3]
```diff
@@ -4,6 +4,7 @@ import Animated, {
   useSharedValue,
   useAnimatedStyle,
   withSpring,
+  SharedValue,
```
Comment (piaskowyk) [inline docs/src/examples/SpringCarousel.tsx:7]:
```suggestion
```

File: docs/src/examples/SpringMassCompare.tsx [hunk 4]
```diff
@@ -5,13 +5,14 @@ import Animated, {
   useAnimatedStyle,
   withSpring,
   withRepeat,
+  SharedValue,
```
Comment (piaskowyk) [inline docs/src/examples/SpringMassCompare.tsx:8]:
```suggestion
```

File: docs/src/examples/TimingEasingCompare.tsx [hunk 5]
```diff
@@ -6,13 +6,14 @@ import Animated, {
   withTiming,
   Easing,
   withRepeat,
+  SharedValue,
```
Comment (piaskowyk) [inline docs/src/examples/TimingEasingCompare.tsx:9]:
```suggestion
```

File: docs/src/examples/UseReducedMotion.tsx [hunk 6]
```diff
@@ -1,6 +1,7 @@
 import React from 'react';
 import { StyleSheet, View } from 'react-native';
 import Animated, {
+  SharedValue,
```
Comment (piaskowyk) [inline docs/src/examples/UseReducedMotion.tsx:4]:
```suggestion
```

File: docs/src/examples/ScrollTo.tsx [hunk 7]
```diff
@@ -5,15 +5,16 @@ import Animated, {
   useDerivedValue,
   useSharedValue,
   scrollTo,
+  SharedValue,
```
Comment (piaskowyk) [inline docs/src/examples/ScrollTo.tsx:8]:
```suggestion
```

File: docs/src/examples/DecayTrain.tsx [hunk 8]
```diff
@@ -9,19 +9,20 @@ import {
   Gesture,
   GestureDetector,
   GestureHandlerRootView,
+  PanGesture,
```
Comment (piaskowyk) [inline docs/src/examples/DecayTrain.tsx:12]: To importing types we should use this syntax:
```suggestion
import type { PanGesture } 'react-native-gesture-handler';
```
Can you apply it also to the rest of occurs?

File: docs/src/examples/AnimatedReactionBasic.jsx
Comment (piaskowyk) [inline docs/src/examples/AnimatedReactionBasic.jsx:1]: Why you removed this file?

File: docs/src/components/InteractiveExample/index.tsx [hunk 9]
```diff
@@ -15,6 +15,56 @@ import ResetDark from '@site/static/img/reset-dark.svg';
 
 import styles from './styles.module.css';
 
+import ts from 'typescript';
+import prettier from 'prettier/standalone';
+import babelParser from 'prettier/parser-babel';
+
+function compileTSXtoJSX(tsxCode: string) {
+  const TEXT_TO_REPLACE_1 = '// 1-COMMENT-TO-REPLACE';
+  const TEXT_TO_REPLACE_2 = '// 2-COMMENT-TO-REPLACE';
```
Comment (piaskowyk) [inline docs/src/components/InteractiveExample/index.tsx:24]: This is unclear for me. Could you explain me what is the purpose of this comments?

**Review by piaskowyk — CHANGES_REQUESTED** [review]

File: docs/src/examples/ScrollTo.tsx (hunk 7 above)
Comment (xnameTM) [reply docs/src/examples/ScrollTo.tsx:8]: this type is used on line 54

File: docs/src/examples/AnimatedReactionBasic.jsx
Comment (xnameTM) [reply docs/src/examples/AnimatedReactionBasic.jsx:1]: cause it was never used

File: docs/src/examples/DecayTrain.tsx (hunk 8 above)
Comment (xnameTM) [reply docs/src/examples/DecayTrain.tsx:12]: yes sir

File: docs/src/components/InteractiveExample/index.tsx (hunk 9 above)
Comment (xnameTM) [reply docs/src/components/InteractiveExample/index.tsx:24]: Yes. The purpose of these comments is to maintain blank spaces between lines. So in my opinion it should stay there.

**Review by piaskowyk — APPROVED** [review]

**Pull request merged**